scripts/python-skywater-pdk/skywater_pdk/cell_list.py

```diff
@@ -0,0 +1,159 @@
+rst_template ="""List of cells
+-------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Cell name
+     - Description
+     - Type
+     - Verilog name
+{cell_list}
+"""
+
+cell_template = """   * - {cell_name}
+     - {description}
+     - {type}
+     - {verilog_name}
+"""
+
+def collect(library_dir) -> Tuple[str, List[str]]:
+    """Collect the available definitions for cells in a library
+
+    Parameters
+    ----------
+    library_dir: str or pathlib.Path
+        Path to a library.
+
+    Returns
+    -------
+    lib : str
+        Library name
+
+    cells : list of pathlib.Path
+        definition files for cells in the library.
+    """
+
+    if not isinstance(library_dir, pathlib.Path):
+        library_dir = pathlib.Path(library_dir)
+
+    libname = None
+    cells = set()
+
+    for p in library_dir.rglob("definition.json"):
+        if not p.is_file():
+            continue
+        cells.add(p)
+        if libname is None:
+            with open(str(p), "r") as sample_json:
+                sample_def = json.load(sample_json)
+                libname = sample_def['library']
+
+    assert len(libname) > 0
+    cells = list(sorted(cells))
+    return libname, cells
+
+
+def generate_rst(library_dir, library_name, cells):
+    """Generate the RST file containing basic information about cells
+
+    Parameters
+    ----------
+    library_dir: str or pathlib.Path
+        Path to a library.
+
+    library_name: str
+        Name of the library
+
+    cells: list of pathlib.Path
+        List of paths to JSON description files
+
+    """
+
+    if not isinstance(library_dir, pathlib.Path):
+        library_dir = pathlib.Path(library_dir)
+
+    file_name = "cell-list.rst"
+
+    cell_list_file = pathlib.Path(library_dir, file_name)
+
+    cell_list = ""
+    for cell in cells:
+        with open(str(cell), "r") as c:
+            cell_json = json.load(c)
+            cell_list = cell_list + cell_template.format(
+                cell_name = cell_json['name'],
+                description = cell_json['description'].replace("\n", "\n       "),
+                type = cell_json['type'],
+                verilog_name = cell_json['verilog_name']
+            )
+
+    try:
+        with(open(str(cell_list_file), "w")) as c:
+            c.write(rst_template.format(
+                cell_list = cell_list
+            ))
+    except FileNotFoundError:
+        print(f"ERROR: Failed to create {str(cell_list_file)}", file=sys.stderr)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+            "library_dir",
+            help="Path to the library.",
+            type=pathlib.Path,
+            nargs=1)
+
+    parser.add_argument(
+            "--debug",
+            help="Include verbose debug output on the console.",
+            action='store_true',
+            default=False)
+
+    args = parser.parse_args()
+    if args.debug:
```

Comment:
Seems like `debug` isn't all that used. Maybe we should get rid of it?

Reply:
Are we ok with the script printing stuff on its own? Or should it be silent?

Reply:
I left unconditional printing, can be removed if needed